iFixit/Veryable
# Feature: Generate Database Docs

## Description

This will be useful in later pull requests for this feature branch when defining a more entangled database schema. 

Prisma has generators that can create different file outputs based on the current schema defined in `schema.prisma`. One of those generators can create a database model language file that can then be uploaded to https://dbdocs.io/docs.  This will make it easier to maintain the documentation for the database since the file will automatically be generated when `npx prisma generate` is ran. 

You can see it here https://dbdocs.io/ardelato/Veryable

**EDIT:** Just realized you need a password to view the project, please ping me if you want to take a look
![DBDocs](https://user-images.githubusercontent.com/22064420/144948508-b5ab08c6-7998-45b0-993e-e8f160fd2f52.png)


## Changes:

- adds a new dev dependency -- prisma dbml generator https://notiz.dev/blog/prisma-dbml-generator
- adds a `docs` folder for future documentation
- adds a `schema.dbml` file


### QA

This simply adds a generator for documentation.

Therefore, you can follow the initial setup steps from https://github.com/iFixit/Veryable/pull/17#issue-1055586897, also making sure to remove the `schema.dbml` file in the `docs` directory before continuing.

Then once you run `npm install` you can run `npx prisma generate` it should generate the `schema.dbml` file in the docs folder. 

Afterward, you can copy and paste the content of the file into https://dbdiagram.io/ and verify it was able to render the database diagram without issues.
